xdr/transaction.go

```diff
@@ -14,3 +14,78 @@ func (tx *Transaction) TimeBounds() *TimeBounds {
 		panic("unsupported transaction type: " + tx.Cond.Type.String())
 	}
 }
+
+// LedgerBounds extracts the ledgerbounds (if any) from the transaction's
+// Preconditions.
+func (tx *Transaction) LedgerBounds() *LedgerBounds {
+	switch tx.Cond.Type {
+	case PreconditionTypePrecondNone:
+		return nil
+	case PreconditionTypePrecondTime:
+		return nil
+	case PreconditionTypePrecondV2:
+		return tx.Cond.V2.LedgerBounds
+	default:
+		panic("unsupported transaction type: " + tx.Cond.Type.String())
```

Comment:
Looks like a copy-paste typo here, the type is preconditions, not transaction:
```suggestion
		panic("unsupported preconditions type: " + tx.Cond.Type.String())
```